quickwit/quickwit-common/src/rendezvous_hasher.rs

```diff
@@ -63,8 +63,8 @@ mod tests {
         let mut socket_set3 = vec![socket1, socket4];
         sort_by_rendez_vous_hash(&mut socket_set3, "key");
 
-        assert_eq!(socket_set1, &[socket1, socket3, socket2, socket4]);
```

Comment:
interesting. is there a change in some hash function?
It might be a good idea to make sure we fix the hash function here.

Reply:
Yeah, I noticed in the commit history this has happened a couple of times before during rust version updates.

It looks like we're using the default hasher from std

```
use std::collections::hash_map::DefaultHasher;
```

I did have a quick scan through the Rust changelog and didn't see any changes specific to hashing but I could have missed something. 

The docs do note though:

> The internal algorithm is not specified, and so it and its hashes should not be relied upon over releases.

Reply:
Yes. Let's stop using the default hasher and "pin" the hash function instead.

Reply:
Let's use a third-party crate, same as Rust: https://crates.io/crates/siphasher.

Reply:
I've updated the PR